File: DayPlan.php
<?php

require_once 'DayOfWeek.php';
require_once 'Meal.php';

final class DayPlan
{
    public function __construct(
        public DayOfWeek $day,
        public Meal $breakfast,
        public Meal $lunch,
        public Meal $tea,
    ) {}
}

File: src/main/java/com/marshall/guy/DayPlan.java
package com.marshall.guy;

public class DayPlan {
    public DayOfWeek day;
    public Meal breakfast;
    public Meal lunch;
    public Meal tea;

    public DayPlan(DayOfWeek day, Meal breakfast, Meal lunch, Meal tea) {
        this.day = day;
        this.breakfast = breakfast;
        this.lunch = lunch;
        this.tea = tea;
    }
}
